lib/ansible/modules/cloud/vmware/vmware_host_sriov.py

```diff
@@ -0,0 +1,387 @@
+
+        if self.num_virt_func > 0:
+            if self.sriov_on is False:
+                self.module.fail_json(
+                    msg="with sriov_on == false,  alowed value for num_virt_func is 0"
+                )
+            params_to_change.update(
+                {"sriovEnabled": True, "numVirtualFunction": self.num_virt_func}
+            )
+
+            # check compatability
+            if not before["sriovCapable"]:
+                self.module.fail_json(
+                    msg="sriov not supported on host= %s, nic= %s" % (hostname, vmnic)
+                )
+
+            if (
+                before["maxVirtualFunctionSupported"]
+                < params_to_change["numVirtualFunction"]
+            ):
+                self.module.fail_json(
+                    msg="maxVirtualFunctionSupported= %d on %s"
+                    % (before["maxVirtualFunctionSupported"], vmnic)
+                )
+
+        # check current settings
+        if before["sriovEnabled"] != params_to_change["sriovEnabled"]:
+            params_to_change["changes"]["sriovEnabled"] = params_to_change[
+                "sriovEnabled"
+            ]
+            params_to_change["change"] = True
+        if before["numVirtualFunction"] != params_to_change["numVirtualFunction"]:
+            if (
+                before["numVirtualFunctionRequested"]
+                != params_to_change["numVirtualFunction"]
+            ):
+                params_to_change["changes"]["numVirtualFunction"] = params_to_change[
+                    "numVirtualFunction"
+                ]
+                params_to_change["change"] = True
+            else:
+                params_to_change["changes"]["msg"] = "Not active (looks not rebooted) "
+
+        if not params_to_change["change"]:
+            msg = (
+                params_to_change["changes"].get("msg", "")
+                + "No any changes, already configured "
+            )
+            params_to_change["changes"]["msg"] = msg
+        return params_to_change
+
+    def set_host_state(self):
+        """Checking and applying ESXi host configuration one by one,
+        from prepared list of hosts in `self.hosts`.
+        For every host applied:
+        - user input checking done via calling `sanitize_params` method
+        - changes applied via calling `_update_sriov` method
+        - host state before and after via calling `_check_sriov`
+        """
+        change_list = []
+        changed = False
+        for host in self.hosts:
+            self.results["before"][host.name] = {}
+            self.results["after"][host.name] = {}
+            self.results["changes"][host.name] = {}
+            self.results["before"][host.name] = self._check_sriov(host)
+            self.results["changes"][host.name]["rebootRequired"] = self.results[
+                "before"
+            ][host.name]["rebootRequired"]
+
+            params_to_change = self.sanitize_params(
```

Comment:
If you choose to make `sanitize_params` and `check_compatibility` separate methods, `sanitize_params` can be called even before the `for loop` above

Reply:
thanks, it will look more clear.